common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -2547,6 +2547,9 @@ public static enum ConfVars {
         "If this config is true only pushed down filters remain in the operator tree, \n" +
         "and the original filter is removed. If this config is false, the original filter \n" +
         "is also left in the operator tree at the original place."),
+    HIVE_JOIN_PUSH_TRANSITIVE_PREDICATES_CONSERVATIVE("hive.optimize.join.transitive.predicates.conservative",
+        false, "Whether to avoid pushing predicates that are hard to simplify. \n"
```

Comment:
I am not sure if we want to keep this so general or focus on what actually happens, which is forbidding disjunction pushdown (e.g., `hive.optimize.join.disjunctive.transitive.predicates.pushdown`).

Regarding the default value maybe it would be better to disallow the optimization by default. From my perspective, correctness is more important than performance so having HS2 crashing with OOM is more serious than having perf regression in some queries. 

Both of the comments above are rather personal preferences so I am not pushing hard for making these changes. The final decision is up to you :)

Reply:
Since we are focusing on the disjunctive predicates (as per another comment), let's opt for `hive.optimize.join.disjunctive.transitive.predicates.pushdown`.

As discussed offline, let's leave the property disabled by default, because it's hard to judge how much benefit/harm there will be for Hive in the wild, we can revisit the default value later on if we have more evidence that this will help.